File: app/src/main/java/com/avdhootsolutions/aswack_shopkeeper/models/BuyVehicle.kt
package com.avdhootsolutions.aswack_shopkeeper.models

import com.google.gson.annotations.SerializedName

class BuyVehicle {

    @SerializedName("category")
    var category: String? = null

    @SerializedName("user_id")
    var user_id: String? = null

    @SerializedName("user_type")
    var user_type: String? = null

    @SerializedName("vehicle_cat")
    var vehicle_cat: String? = null

    @SerializedName("vehicle_brand")
    var vehicle_brand: ArrayList<String> = ArrayList<String>()

    @SerializedName("vehicle_type")
    var vehicle_type: ArrayList<String> = ArrayList<String>()

    @SerializedName("vehicle_model")
    var vehicle_model: ArrayList<String> = ArrayList<String>()

    @SerializedName("vehicle_year")
    var vehicle_year: ArrayList<String> = ArrayList<String>()

    @SerializedName("vehicle_fuel")
    var vehicle_fuel: ArrayList<String> = ArrayList<String>()

    @SerializedName("transmission")
    var transmission: String? = null



}

File: lib/models/buy_vehicle.dart
class BuyVehicle {
  String? category;
  String? userId;
  String? userType;
  String? vehicleCat;
  List<String> vehicleBrand = [];
  List<String> vehicleType = [];
  List<String> vehicleModel = [];
  List<String> vehicleYear = [];
  List<String> vehicleFuel = [];
  String? transmission;

  Map<String, dynamic> toJson() => {
        'category': category,
        'user_id': userId,
        'user_type': userType ?? '1', // 1 = SELLER
        'vehicle_cat': vehicleCat,
        'vehicle_brand': vehicleBrand,
        'vehicle_type': vehicleType,
        'vehicle_model': vehicleModel,
        'vehicle_year': vehicleYear,
        'vehicle_fuel': vehicleFuel,
        'transmission': transmission ?? '',
      };
}
